### PR TITLE
Refactor based on latest XO recommendations

### PR DESCRIPTION
**Pull request by XhmikosR**
Split from my xo branch. These are mostly autofixes.

Non-whitespace diff: https://github.com/stylelint/stylelint/pull/5546/files?w=1

### CODE REVIEW

File: lib/rules/color-named/index.js [hunk 1]
```diff
@@ -113,8 +113,8 @@ const rule = (primary, secondaryOptions) => {
 					const normalizedFunctionString = valueParser.stringify(node).replace(/\s+/g, '');
 					let namedColor;
 
-					for (let i = 0, l = namedColors.length; i < l; i++) {
-						namedColor = namedColors[i];
+					for (const namedColor_ of namedColors) {
+						namedColor = namedColor_;
```
Comment (jeddy3) [inline lib/rules/color-named/index.js:117]: Is this right?

**Review by jeddy3 — COMMENTED** [review]
Thanks!

Only one query, otherwise LGTM.

File: lib/rules/color-named/index.js (hunk 1 above)
Comment (XhmikosR) [reply lib/rules/color-named/index.js:117]: TBH this baffled me too. It seems that `namedColor` is shared between the two loops. Maybe they could/should be combined later?

**Review by ybiquitous — COMMENTED** [review]
Thanks, mostly LGTM 👍🏼 

I think some type annotations are no longer required thanks to the idiom using `flat()`.

**Comment by XhmikosR** [comment]
@ybiquitous can you push your changes please? I'm not familiar with TS almost at all :)

File: lib/rules/color-named/index.js (hunk 1 above)
Comment (ybiquitous) [reply lib/rules/color-named/index.js:117]: I think this PR is good timing to combine them! 👍🏼

**Comment by ybiquitous** [comment]
OK, just a minute. 👌🏼

**Comment by ybiquitous** [comment]
Done. bf6f4abb

File: lib/rules/color-named/index.js (hunk 1 above)
Comment (XhmikosR) [reply lib/rules/color-named/index.js:117]: Feel free to push here :) Or someone could do it in another PR later

Comment (ybiquitous) [reply lib/rules/color-named/index.js]: Fixed! 9ae83efa

**Review by ybiquitous — APPROVED** [review]
@XhmikosR Thank you for cleaning the code! 👏🏼

**Comment by XhmikosR** [comment]
NP, thanks for the help :) I hope nothing broke because I did hit a wrong autofix here: https://github.com/stylelint/stylelint/blob/5c55141f53b62e0d452285fc4460e2ae0fe710c6/lib/rules/no-extra-semicolons/index.js#L30 which resulted in:

```
TypeError: string.entries is not a function or its return value is not iterable
```

**Comment by ybiquitous** [comment]
I see. `String` does not have `entries()` method, not like `Array`. I think it is correct not to auto-fix. 😃

**Comment by XhmikosR** [comment]
Yeah, it's just that it couldn't detect the type hence the wrong fix. Too lazy to report it...

I think the rest `for...of` changes are OK, though.

File: lib/__tests__/disableRanges.test.js [hunk 2]
```diff
@@ -990,8 +990,8 @@ function expectDisableRanges(result, expected) {
 
 		expect(actualRanges).toHaveLength(expectedRanges.length);
 
-		for (let i = 0; i < expectedRanges.length; i++) {
-			expectDisableRange(actualRanges[i], expectedRanges[i]);
+		for (const [i, expectedRange] of expectedRanges.entries()) {
```
Comment (alexander-akait) [inline lib/__tests__/disableRanges.test.js:993]: Just note - simple `for` is faster then `for/of` in many cases

**Comment by ybiquitous** [comment]
Perhaps `Object.entries()` may be available like this, but I think it is good to do it with another pull request. 😃 

```js
for (const [i, char] of Object.entries(string)) { ... }
```

File: lib/__tests__/disableRanges.test.js (hunk 2 above)
Comment (alexander-akait) [reply lib/__tests__/disableRanges.test.js:993]: But it is very small optimizations and most likely will not affect anything, excluding parts that really need to be optimized

Comment (XhmikosR) [reply lib/__tests__/disableRanges.test.js:993]: Yeah, I'm aware of that, but for of is used already in plenty of places in the codebase.

```
C:\Users\xmr\Desktop>matcha bench-loop.js

        125,000,000 ops/sec > for (post) (67.5x)
        110,000,000 ops/sec > for (pre) (59.5x)
        111,000,000 ops/sec > for (cached length) (59.9x)
        111,000,000 ops/sec > forEach (60.1x)
         76,000,000 ops/sec > for of (41.2x)
          2,190,000 ops/sec > for in (1.19x)
          1,850,000 ops/sec > for in (own) (1x)

  Benches: 7
  Fastest: for (post)
  Elapsed: 38.8s
```

That being said, I don't think this is a real issue and readability plus the fact that you can have off by one errors is better :)

**Review by jeddy3 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**